### DISCUSSION TITLE
mkdocs material to support MDX

### DISCUSSION THREAD

**Discussion by jbesomi** [squidfunk/mkdocs-material#3621] [Ideas]
Hey @squidfunk, first of all, congratulations on your project, fully love it: simple, beautiful, and powerful.
It would be very great if we could add support for [MDX](https://mdxjs.com/). This would allow to add JSX components and reuse the code. I read somewhere you loved it and thought about adding it as a feature ... but I couldn't find anything related to that, neither in the `mkdocs` website.
  1. Is that something you are still considering adding? You haven't mentioned it in the [2022 roadmap](https://squidfunk.github.io/mkdocs-material/blog/2021/the-past-present-and-future/)
  2. How hard is it in your view to support that feature (in terms of code/logic changes and working hours)?
  3. Would you like some help to add it? More than happy to contribute, as I need that feature anyway!
  
  Looking forward to hearing both Martin's opinion and the feedback from all the `mkdocs material` community! 🤗
  Thanks in advance for your consideration!

**Comment by squidfunk** [comment]
Thanks for suggesting. Yes, I'm a fan of MDX. However, MDX leverages JSX, which is primarily targeted at single-page applications. Yes, we also use JSX to render components via JavaScript, but MDX would mean that the Markdown written by the author would need to be processed not only to static HTML but the JSX markup would need to be extracted and both would somehow need to be put back together when viewing the page. I can't see how this could be pulled off in a degree that solves the problem at scale, i.e. for all users and not just for a select number of special cases. Also, remember that Material for MkDocs generates a static site. It adds interactivity via JavaScript, but the rendering is all through HTML except for some exceptions like the version selector or search. Furthermore, adding MDX capability would not only need support by the theme, but by the Markdown parser. Thus, it would need to be implemented by Python Markdown or some other upstream project that delivers the final Markdown to be rendered to Material for MkDocs, which after all is "just a theme".

So to answer your questions:

1. Is that something you are still considering adding? You haven't mentioned it in the [2022 roadmap](https://squidfunk.github.io/mkdocs-material/blog/2021/the-past-present-and-future/):
   __No.__

2. How hard is it in your view to support that feature (in terms of code/logic changes and working hours)?
    __Very hard, as it's not a good fit with the Python Markdown / MkDocs pipeline.__

3. Would you like some help to add it? More than happy to contribute, as I need that feature anyway!
    __I'm not up for maintaining this feature, so no plans for merging. I believe it could be a great third-party extension though.__

**Reply by hasansezertasan** [reply]
This is the best, most reasonable, and encouraging answer I've seen on GitHub so far.

**Comment by jbesomi** [comment]
Hey Martin, thank you very much for your exhaustive reply. Make complete sense!
